src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -2067,6 +2067,61 @@ public override void M<U>(U t)
                 );
         }
 
+        [Fact, WorkItem(47513, "https://github.com/dotnet/roslyn/issues/47513")]
+        public void BothGetHashCodeAndEqualsAreDefined()
+        {
+            var src = @"
+public sealed record C
+{
+    public object Data;
+    public bool Equals(C c) { return false; }
+    public override int GetHashCode() { return 0; }
+}";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(47513, "https://github.com/dotnet/roslyn/issues/47513")]
+        public void BothGetHashCodeAndEqualsAreNotDefined()
+        {
+            var src = @"
+public sealed record C
+{
+    public object Data;
+}";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(47513, "https://github.com/dotnet/roslyn/issues/47513")]
+        public void GetHashCodeIsDefinedButEqualsIsNot()
+        {
+            var src = @"
+public sealed record C
+{
+    public object Data;
+    public override int GetHashCode() { return 0; }
+}";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(47513, "https://github.com/dotnet/roslyn/issues/47513")]
+        public void EqualsIsDefinedButGetHashCodeIsNot()
+        {
+            var src = @"
+public sealed record C
+{
+    public object Data;
+    public bool Equals(C c) { return false; }
+}";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (5,17): warning CS8851: 'C' defines 'Equals' but not 'GetHashCode'
+                //     public bool Equals(C c) { return false; }
+                Diagnostic(ErrorCode.WRN_RecordEqualsWithoutGetHashCode, "Equals").WithArguments("C").WithLocation(5, 17));
+        }
```

Comment:
Can you add a test where `Equals(object other)` is defined, both with and without also defining `Equals(C c)`?

Reply:
I think defining Equals(object) is an error in records, which should have already been tested.

Reply:
I should definitely be an error, but I'd still like to have tests of the interaction with this new warning.

Reply:
@333fred I can't get how this can be different from the existing test?

https://github.com/dotnet/roslyn/blob/c2d294c6eac897aa8ac9390d23aeb368d9c9ce3c/src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs#L221-L277

The existing test shows that the new warning isn't produced. (It wouldn't if the new warning had level >= 5, but that's not the case)

Reply:
Ah, I must have missed that test. Thansk